src/v/cluster/topics_frontend.cc

```diff
@@ -133,10 +150,19 @@ ss::future<std::vector<topic_result>> topics_frontend::update_topic_properties(
             co_return create_topic_results(updates, map_errc(result.error()));
         }
 
-        co_return co_await ssx::parallel_transform(
+        auto results = co_await ssx::parallel_transform(
           std::move(updates), [this, timeout](topic_properties_update update) {
               return do_update_topic_properties(std::move(update), timeout);
           });
+
+        // we are not really interested in the result comming from the
+        // linearizable barrier, results comming from the previous steps will be
+        // propagated to clients
+        if (has_any_success(results)) {
+            co_await stm_linearizable_barrier(timeout).discard_result();
```

Comment:
Write that `this is just an optimization, this doesn't affect correctness of the protocol`